### DISCUSSION TITLE
Do you think this could be used for SAML SSO?

### DISCUSSION THREAD

**Discussion by kishba** [sergiodxa/remix-auth#91] [Q&A] [answered]
I come from a Rails background, but Remix has me curious about building with JavaScript. One project that I could probably use for a proof of concept currently supports SAML for single sign on. 

Is supporting SAML outside the goals or intended use of remix-auth?

I noticed that Passport has a strategy named [passport-saml](https://github.com/node-saml/passport-saml) -- is it possible to use that as a strategy in remix-auth?

**Comment by kishba** [comment]
I stumbled upon [samlify](https://samlify.js.org)'s [repo](https://github.com/tngan/samlify) which looks like something I could use within a custom remix-auth strategy. I appreciate any advice if anyone else has tried this before!

**Comment by sergiodxa** [comment, marked as answer]
I’ve never used SAML before but if there’s a Passport Strategy for SAML then you should be able to port it to Remix Auth.

**Comment by deepakprabhakara** [comment]
Hi @kishba, did you get around to trying it with samlify? We have just released an example with Remix + SAML (using our open source service called SAML Jackson). We are also checking to see if it can be embedded as an npm library so you do not have to run an additional service. You can check it out the remix-auth strategy here - https://github.com/boxyhq/remix-auth-saml. And a demo application is here - https://github.com/boxyhq/jackson-remix-auth. What we do differently here with SAML is that we have given it an OAuth 2.0 wrapper so it blends right in. Cc: @sergiodxa

**Reply by kishba** [reply]
Sorry I missed your reply back in March! I'm back to looking for a SAML solution, and this is really cool. I'm going to try out SAML Jackson.

**Reply by deepakprabhakara** [reply]
@kishba Thank you, looking forward to your feedback. Please have a look at https://boxyhq.com/guides/jackson/frameworks/remix, any feedback on improving the example is highly appreciated.

**Comment by christopherpickering** [comment]
Howdy, I made a demo Remix site using the blues stack that incorporates the samlify package for SAML sso without middleman tools.

https://github.com/christopherpickering/remix-saml-authentication
